assets/js/modules/adsense/common/use-snippet-switch.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * AdSense Use Snippet Switch component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import Switch from '../../../components/switch';
+import SettingsNotice from '../../../components/settings-notice';
+import { trackEvent } from '../../../util';
+import { STORE_NAME } from '../datastore';
+const { useSelect, useDispatch } = Data;
+
+export default function UseSnippetSwitch( props ) {
+	const {
+		label = __( 'Let Site Kit place code on your site', 'google-site-kit' ),
+		checkedMessage,
+		uncheckedMessage,
```

Comment:
Why pass these as props?

Reply:
These messages are based on the context of where we display this component. While we could generally handle existing tag cases in here potentially and update the message, there is another nuance that users should see different messaging based on whether an account was already approved before setting up with Site Kit or whether it got approved at some point during setup in Site Kit. I think that is way too specific to bake into this generally reusable component, and the copy from setup also doesn't fully apply to settings.